From 2f09c8f092cc871178bebbf84acfcb1c2c2f332f Mon Sep 17 00:00:00 2001
From: Jack Yu <jackyu@google.com>
Date: Fri, 22 Jul 2016 15:39:09 -0700
Subject: [PATCH 1/9] Fixed that emergency messages are not enabled for
 Japanese carriers

The emergency ETWS (Earthquake and Tsunami Warning System) other
messages were not enabled during boot up. Added the corresponding
channel 4356 (0x1104) into the configuration list.

bug: 30295364
Change-Id: Ib9d0765d418b26b38c0f5047039c5c7ea9fbd714
(cherry picked from commit bc0c1a983dd8d83d5c84afe4a2fa1f3ae09bed42)
---
 .../android/cellbroadcastreceiver/CellBroadcastConfigService.java | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/com/android/cellbroadcastreceiver/CellBroadcastConfigService.java b/src/com/android/cellbroadcastreceiver/CellBroadcastConfigService.java
index 451f9a7..a7273fb 100644
--- a/src/com/android/cellbroadcastreceiver/CellBroadcastConfigService.java
+++ b/src/com/android/cellbroadcastreceiver/CellBroadcastConfigService.java
@@ -344,12 +344,18 @@ public class CellBroadcastConfigService extends IntentService {
 
         /** Enable GSM ETWS series messages. */
 
-        // Enable/Disable GSM ETWS messages.
+        // Enable/Disable GSM ETWS messages (4352 ~ 4354).
         setCellBroadcastRange(manager, enableEtwsAlerts,
                 SmsManager.CELL_BROADCAST_RAN_TYPE_GSM,
                 SmsCbConstants.MESSAGE_ID_ETWS_EARTHQUAKE_WARNING,
                 SmsCbConstants.MESSAGE_ID_ETWS_EARTHQUAKE_AND_TSUNAMI_WARNING);
 
+        // Enable/Disable GSM ETWS other messages (4356).
+        setCellBroadcastRange(manager, enableEtwsAlerts,
+                SmsManager.CELL_BROADCAST_RAN_TYPE_GSM,
+                SmsCbConstants.MESSAGE_ID_ETWS_OTHER_EMERGENCY_TYPE,
+                SmsCbConstants.MESSAGE_ID_ETWS_OTHER_EMERGENCY_TYPE);
+
         // Enable/Disable GSM ETWS test messages (4355).
         setCellBroadcastRange(manager, enableEtwsTestAlerts,
                 SmsManager.CELL_BROADCAST_RAN_TYPE_GSM,
-- 
2.10.0

